### PR TITLE
Fix reference color in framebufferblit test

### PR DESCRIPTION
**Pull request by qkmiao**
Fix two main problems in current code:
1. Set alpha channel as 1 instead of 255 for RGB format.
2. Don not need to clamp float format color to [0, 1]. The value is
always in [valueMin, valueMax].

### CODE REVIEW

File: sdk/tests/deqp/functional/gles3/es3fFramebufferBlitTests.js [hunk 1]
```diff
@@ -722,14 +722,16 @@ goog.scope(function() {
...
+        for (var ndx = 0; ndx < 2; ndx++) {
+            ctx.bindFramebuffer(gl.FRAMEBUFFER, ndx ? dstFbo : srcFbo);
+            if (ndx) {
+                gradientToDstShader.setGradient(ctx, gradShaderDstID, dstRangeInfo.valueMax, dstRangeInfo.valueMin);
+                rrUtil.drawQuad(ctx, gradShaderDstID, [-1, -1, 0], [1, 1, 0]);
+            } else {
+                gradientToSrcShader.setGradient(ctx, gradShaderSrcID, srcRangeInfo.valueMin, srcRangeInfo.valueMax);
+                rrUtil.drawQuad(ctx, gradShaderSrcID, [-1, -1, 0], [1, 1, 0]);
+            }
+        }
```
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fFramebufferBlitTests.js]: https://android.googlesource.com/platform/external/deqp/+/deqp-dev/modules/gles3/functional/es3fFramebufferBlitTests.cpp#398

File: sdk/tests/deqp/functional/gles3/es3fFramebufferBlitTests.js [hunk 2]
```diff
@@ -722,14 +722,16 @@ goog.scope(function() {
...
-        ctx.bindFramebuffer(gl.FRAMEBUFFER, dstFbo);
-        gradientToSrcShader.setGradient(ctx, gradShaderSrcID, srcRangeInfo.valueMin, dstRangeInfo.valueMax);
-        rrUtil.drawQuad(ctx, gradShaderSrcID, [-1, -1, 0], [1, 1, 0]);
+        for (var ndx = 0; ndx < 2; ndx++) {
+            ctx.bindFramebuffer(gl.FRAMEBUFFER, ndx ? dstFbo : srcFbo);
+            if (ndx) {
+                gradientToDstShader.setGradient(ctx, gradShaderDstID, dstRangeInfo.valueMax, dstRangeInfo.valueMin);
+                rrUtil.drawQuad(ctx, gradShaderDstID, [-1, -1, 0], [1, 1, 0]);
+            } else {
+                gradientToSrcShader.setGradient(ctx, gradShaderSrcID, srcRangeInfo.valueMin, srcRangeInfo.valueMax);
```
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fFramebufferBlitTests.js]: C++ should be wrong here: dstRangeInfo.valueMax is used. But it doesn't matter. dstRangeInfo.valueMax and srcRangeInfo.valueMax have the same value.

File: sdk/tests/deqp/functional/gles3/es3fFramebufferBlitTests.js [hunk 3]
```diff
@@ -722,14 +722,16 @@ goog.scope(function() {
...
-
-        rrUtil.drawQuad(ctx, gradShaderDstID, [-1, -1, 0], [1, 1, 0]);
-
-        ctx.bindFramebuffer(gl.FRAMEBUFFER, dstFbo);
-        gradientToSrcShader.setGradient(ctx, gradShaderSrcID, srcRangeInfo.valueMin, dstRangeInfo.valueMax);
-        rrUtil.drawQuad(ctx, gradShaderSrcID, [-1, -1, 0], [1, 1, 0]);
+        for (var ndx = 0; ndx < 2; ndx++) {
+            ctx.bindFramebuffer(gl.FRAMEBUFFER, ndx ? dstFbo : srcFbo);
+            if (ndx) {
+                gradientToDstShader.setGradient(ctx, gradShaderDstID, dstRangeInfo.valueMax, dstRangeInfo.valueMin);
```
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fFramebufferBlitTests.js]: I think reverse order of Max and Min are intentional here.

File: sdk/tests/deqp/functional/gles3/es3fFboTestUtil.js [hunk 4]
```diff
@@ -283,7 +283,7 @@ es3fFboTestUtil.FboIncompleteException.prototype.getReason = function() {return
             /** @const {number} */ var f1 = 0.5 + (x - y) * 0.5;
             /** @const {Array<number>} */ var fv = [f0, f1, 1.0 - f0, 1.0 - f1];
 
-            /** @const {Array<number>} */ var color = deMath.clampVector(deMath.add(gradientMin, deMath.multiply(deMath.subtract(gradientMax, gradientMin), fv)), 0, 1);
+            /** @const {Array<number>} */ var color = deMath.add(gradientMin, deMath.multiply(deMath.subtract(gradientMax, gradientMin), fv));
```
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fFboTestUtil.js]: https://android.googlesource.com/platform/external/deqp/+/deqp-dev/modules/gles3/functional/es3fFboTestUtil.cpp#222
color is always in [min, max] and needn't to clamp to [0, 1]. The same below.

File: sdk/tests/deqp/framework/referencerenderer/rrUtil.js [hunk 5]
```diff
@@ -94,6 +97,12 @@ goog.scope(function() {
         ctx.deleteBuffer(bufIDs[0]);
         ctx.deleteBuffer(bufIDs[1]);
         ctx.deleteVertexArray(vaoID);
+
+        if (posLoc >= 0)
+          ctx.disableVertexAttribArray(posLoc);
+
+        if (coordLoc >= 0)
+          ctx.disableVertexAttribArray(coordLoc);
```
Comment (qkmiao) [inline sdk/tests/deqp/framework/referencerenderer/rrUtil.js]: https://android.googlesource.com/platform/external/deqp/+/deqp-dev/framework/opengl/simplereference/sglrContextUtil.cpp#152

File: sdk/tests/deqp/framework/common/tcuTexture.js [hunk 6]
```diff
@@ -1365,6 +1365,7 @@ tcuTexture.ConstPixelBufferAccess.prototype._getPixelInternal = function(x, y, z
             color[0] /= 255;
             color[1] /= 255;
             color[2] /= 255;
+            color[3] = 1;
```
Comment (qkmiao) [inline sdk/tests/deqp/framework/common/tcuTexture.js]: https://android.googlesource.com/platform/external/deqp/+/deqp-dev/framework/common/tcuTexture.cpp#52

**Comment by qkmiao** [comment]
This fixes all tests in framebufferblit except in depth_stencil.html. It also fixes new added float type tests in fborender/resize_o3.html 
FYI: @Jiawei-Shao .

**Comment by zhenyao** [comment]
Great work!

**Pull request merged**